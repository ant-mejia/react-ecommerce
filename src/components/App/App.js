import React, { Component } from 'react';
import classnames from 'classnames';
import { Link } from 'react-router-dom';
import logo from './logo.svg';
// import './style.css';

class App extends Component {

  render() {
    const { className, ...props } = this.props;
    return (
      <div className={classnames('App', className)} {...props}>
        <h1>Welcome to React!!</h1>
      </div>
    );
  }
}

export default App;
