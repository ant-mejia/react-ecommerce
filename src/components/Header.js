import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Header extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <header ref="header" className='c-header'>
        <div className="c-header-menu u-fixed">
          <div ref="headerMenuButton" onClick={this.props.toggleHeader} className="c-header-menu_button">
            <span/>
          </div>
          {/* <div>
            <img src="https://cdn.dribbble.com/assets/logo-bw-0200c7483844c355752e89efaa4ba89b83c9c591d70254ba10f4b25d901359d0.gif"/>
          </div> */}
        </div>
        <div className="c-header-container">
          <div className="c-header-container_menu">
            <div className="c-header-container_menu-wrapper">
              <Link className="c-header-container_menu-item" to="/index">Home</Link>
              <Link className="c-header-container_menu-item" to="/index">Home</Link>
              <Link className="c-header-container_menu-item" to="/index">Home</Link>
              <Link className="c-header-container_menu-item" to="/index">Home</Link>
              <Link className="c-header-container_menu-item" to="/index">Home</Link>
              <Link className="c-header-container_menu-item" to="/index">Home</Link>
              <div className="c-header-container_menu-item-alternate-wrapper">
                <Link to="/login"><p className="c-header-container_menu-item-alternate-item">Sign In</p></Link>
                <Link to="/register"><p className="c-header-container_menu-item-alternate-item">Register</p></Link>
                <p className="c-header-container_menu-item-alternate-item">Link</p>
              </div>
              <div className="c-header-container_menu-footer">
                This is a really, really, really, really, really, really, really, really, really, really, really, long sentence
              </div>
            </div>
          </div>
          <div className="c-header-container_page-stack"></div>
        </div>
      </header>
    );
  }

}

export default Header;