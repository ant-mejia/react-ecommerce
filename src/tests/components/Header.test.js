import React from 'react';
import { Link } from 'react-router-dom';
import { shallow, mount, render } from 'enzyme';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import Header from '../../components/Header';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../../app'
configure({ adapter: new Adapter() });


describe('Header Component - Lifecycle', () => {
  const renderedComponent = (props = {}) => shallow(<Header {...props} />);
  const app = new App();
  const props = {
    actions: app.actions()
  };
  it('Renders correctly without errors', () => {
    expect(toJson(renderedComponent(props))).toMatchSnapshot();
  });
});

describe('Header Component - Events', () => {
  const renderedComponent = (props = {}) => shallow(<Header {...props} />);
  const app = new App();
  const props = {
    actions: app.actions()
  };
  it('simulates click events', () => {
    const onButtonClick = jest.fn();
    const wrapper = shallow(<Header onClick={onButtonClick} />);
    wrapper.find('button').simulate('click');
    expect(onButtonClick.calledOnce).to.equal(true);
  });
});