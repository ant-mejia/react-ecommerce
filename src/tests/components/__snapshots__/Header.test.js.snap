// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component - Lifecycle Renders correctly without errors 1`] = `
<header
  className="c-header"
>
  <div
    className="c-header-menu u-fixed"
  >
    <div
      className="c-header-menu_button"
    >
      <span />
    </div>
  </div>
  <div
    className="c-header-container"
  >
    <div
      className="c-header-container_menu"
    >
      <div
        className="c-header-container_menu-wrapper"
      >
        <Link
          className="c-header-container_menu-item"
          replace={false}
          to="/index"
        >
          Home
        </Link>
        <Link
          className="c-header-container_menu-item"
          replace={false}
          to="/index"
        >
          Home
        </Link>
        <Link
          className="c-header-container_menu-item"
          replace={false}
          to="/index"
        >
          Home
        </Link>
        <Link
          className="c-header-container_menu-item"
          replace={false}
          to="/index"
        >
          Home
        </Link>
        <Link
          className="c-header-container_menu-item"
          replace={false}
          to="/index"
        >
          Home
        </Link>
        <Link
          className="c-header-container_menu-item"
          replace={false}
          to="/index"
        >
          Home
        </Link>
        <div
          className="c-header-container_menu-item-alternate-wrapper"
        >
          <Link
            replace={false}
            to="/login"
          >
            <p
              className="c-header-container_menu-item-alternate-item"
            >
              Sign In
            </p>
          </Link>
          <Link
            replace={false}
            to="/register"
          >
            <p
              className="c-header-container_menu-item-alternate-item"
            >
              Register
            </p>
          </Link>
          <p
            className="c-header-container_menu-item-alternate-item"
          >
            Link
          </p>
        </div>
        <div
          className="c-header-container_menu-footer"
        >
          This is a really, really, really, really, really, really, really, really, really, really, really, long sentence
        </div>
      </div>
    </div>
    <div
      className="c-header-container_page-stack"
    />
  </div>
</header>
`;
